lib/saputils.pm

```diff
@@ -105,50 +187,52 @@ sub calculate_hana_topology {
 
 =item B<input> - return value of calculate_hana_topology
 
-=item B<node_state_match> - string used to match the online state in field node_state. Default 'online'
+=item B<node_state_match> - string used to match the online state in field node_state.
 
 =back
 =cut
 
 
 sub check_hana_topology {
     my (%args) = @_;
-    croak("Argument <input> missing") unless $args{input};
+    croak("check_hana_topology [ERROR] Argument <input> missing") unless $args{input};
     my $topology = $args{input};
-    $args{node_state_match} //= 'online';
+    my $node_state_match = ($args{node_state_match} eq 'online' or $args{node_state_match} =~ /[1-9]+/) ? '4' : '1';
 
     my $all_online = 1;
     my $prim_count = 0;
     my $sok_count = 0;
-
-    foreach my $host (keys %$topology) {
+    foreach my $site (keys %{$topology->{Site}}) {
         # first check presence of all fields needed in further tests.
         # If something is missing the topology is considered invalid.
-        foreach (qw(node_state sync_state)) {
-            unless (defined($topology->{$host}->{$_})) {
-                record_info('check_hana_topology', "Missing '$_' field in topology output for host $host");
+        foreach (qw(lss srPoll)) {
+            unless (defined($topology->{Site}->{$site}->{$_})) {
+                print('check_hana_topology', ' [ERROR] ', "Missing '$_' field in topology output for site $site");
                 return 0;
             }
         }
 
         # Check node_state
-        if ($topology->{$host}->{node_state} !~ $args{node_state_match}) {
-            record_info('check_hana_topology', "node_state: '$topology->{$host}->{node_state}' is not '$args{node_state_match}' for host $host");
+        if ($topology->{'Site'}->{$site}->{'lss'} ne $node_state_match) {
```

Comment:
I would suggest creating a variable which contains `$topology->{'Site'}->{$site}->{'lss'}` and use just the variable instead of copying the whole line multiple times.
Makes your code more readable and if there is a change in the future, you don't have to search for all occasions and miss one by chance. 
Also hash keys do not need to be quoted usually.

Example: 
```
my $site_lss = $topology->{Site}->{$site}->{lss};
my $site_mns = $topology->{Site}->{$site}->{mns};
if ($site_lss ne $node_state_match) {
  print('check_hana_topology', ' [ERROR] ', "node_state: $site_lss is not $node_state_match for host $site_mns \n");
```